tests/src/Unit/PermissionEventTest.php

```diff
@@ -0,0 +1,461 @@
+   *   An array of test permissions.
+   * @param string $entity_type_id
+   *   The entity type ID of the group type to which the permissions apply.
+   * @param string $bundle_id
+   *   The bundle ID of the group type to which the permissions apply.
+   *
+   * @covers ::hasPermission
+   *
+   * @dataProvider permissionsProvider
+   */
+  public function testHasPermission($permissions, $entity_type_id, $bundle_id) {
+    $event = new PermissionEvent($entity_type_id, $bundle_id);
+
+    foreach ($permissions as $name => $permission) {
+      $this->assertFalse($event->hasPermission($name));
+      $event->setPermission($name, $permission);
+      $this->assertTrue($event->hasPermission($name));
+    }
+  }
+
+  /**
+   * @param array $permissions
+   *   An array of test permissions.
+   * @param string $entity_type_id
+   *   The entity type ID of the group type to which the permissions apply.
+   * @param string $bundle_id
+   *   The bundle ID of the group type to which the permissions apply.
+   *
+   * @covers ::getEntityTypeId
+   *
+   * @dataProvider permissionsProvider
+   */
+  public function testGetEntityTypeId($permissions, $entity_type_id, $bundle_id) {
+    $event = new PermissionEvent($entity_type_id, $bundle_id);
+    $this->assertEquals($entity_type_id, $event->getEntityTypeId());
+  }
+
+  /**
+   * @param array $permissions
+   *   An array of test permissions.
+   * @param string $entity_type_id
+   *   The entity type ID of the group type to which the permissions apply.
+   * @param string $bundle_id
+   *   The bundle ID of the group type to which the permissions apply.
+   *
+   * @covers ::getBundleId
+   *
+   * @dataProvider permissionsProvider
+   */
+  public function testGetBundleId($permissions, $entity_type_id, $bundle_id) {
+    $event = new PermissionEvent($entity_type_id, $bundle_id);
+    $this->assertEquals($bundle_id, $event->getBundleId());
+  }
+
+  /**
+   * @param array $permissions
+   *   An array of test permissions.
+   * @param string $entity_type_id
+   *   The entity type ID of the group type to which the permissions apply.
+   * @param string $bundle_id
+   *   The bundle ID of the group type to which the permissions apply.
+   *
+   * @covers ::filterByDefaultRole
+   *
+   * @dataProvider permissionsProvider
+   */
+  public function testFilterByDefaultRole($permissions, $entity_type_id, $bundle_id) {
+    $event = new PermissionEvent($entity_type_id, $bundle_id);
+    $event->setPermissions($permissions);
+
+    $default_roles = [
+      OgRoleInterface::ANONYMOUS,
+      OgRoleInterface::AUTHENTICATED,
+      OgRoleInterface::ADMINISTRATOR,
+    ];
+    foreach ($default_roles as $default_role) {
+      $expected = array_filter($permissions, function ($permission) use ($default_role) {
+        return !empty($permission['default roles']) && in_array($default_role, $permission['default roles']);
+      });
+      $this->assertEquals($expected, $event->filterByDefaultRole($default_role));
```

Comment:
This is one thing I am not overly happy with, I am using the same logic in the test as in the implementation. But since the alternative is to provide a huge dataProvider with an array of expected outcomes for each role, I settled for this. It is too trivial to spend a lot of time on.